src/api/proxy/CacheProxy.ts

```diff
@@ -0,0 +1,70 @@
+import NodeCache from 'node-cache'
+
+export class CacheProxy<T> {
+  protected api: T
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private cache: NodeCache
+
+  public constructor(api: T) {
+    this.api = api
+    this.cache = new NodeCache({ useClones: false })
+  }
+
+  private get<R>(hash: string): R | undefined {
+    const obj = this.cache.get(hash)
+
+    if (obj) {
+      console.debug(`cache hit for ${hash}`)
+      return obj as R
+    }
+
+    return
+  }
```

Comment:
could be rewritten as:

```ts
    ....
    if (!obj) return

    console.debug(`cache hit for ${hash}`)
    return obj
  }
```

to avoid `as blah` ts code

Reply:
I was missing the type on this.get()

But applying your suggestion regardless, looks better